packages/@ember/-internals/glimmer/lib/views/outlet.ts

```diff
@@ -57,6 +57,7 @@ export default class OutletView {
         outlet: TOP_LEVEL_OUTLET,
         name: TOP_LEVEL_NAME,
         controller: undefined,
+        model: undefined,
```

Comment:
FWIW, I'd prefer to make this more generic in the internals. As mentioned above, `@ember/test-helpers` uses outlet state to support rendering tests, and it would be quite nice to allow them to use named arguments (instead of requiring `this.foo` from the test context). It seems that making this structure a tad more generic (but preventing routes from using it) would allow that.

Reply:
seems like a good future improvement 😄

(It is not _exactly_ trivial to translate this to be more generic either, due to the way things are implemented)

Reply:
Ya, totally agree this isn't a blocker, but I would love to discuss this sometime with you (hopefully while the issue is still somewhat fresh in your mind).